wasm/main.py

```diff
@@ -2811,40 +2801,39 @@ def spec_externtype_matching_limits(limits_a: Limits, limits_b: Limits) -> str:
 def spec_externtype_matching(externtype1, externtype2):
     logger.debug('spec_externtype_matching(%s, %s)', externtype1, externtype2)
 
-    if externtype1[0] != externtype2[0]:
-        raise Unlinkable(f"Mismatch in extern types: {externtype1[0]} != {externtype2[0]}")
-    elif "func" == externtype1[0] and "func" == externtype2[0]:
-        if externtype1[1] == externtype2[1]:
+    if type(externtype1) is not type(externtype2):
+        raise Unlinkable(
+            f"Mismatch in extern types: {type(externtype1)} != {type(externtype2)}"
+        )
+    elif isinstance(externtype1, FuncType) and isinstance(externtype2, FuncType):
```

Comment:
Hm, checking them both after showing that they are equal feels redundant. No objection really, but I'd be happy with skipping the second check, too.